Issue #3900: Split and Organize Checkstyle inputs by Test for TypecastParenPad

ps-sp opened the pull request:
Issue #3900 : Modified TypecastParenPadCheckTest.java to accommodate for the change in the location of test-resources the corresponding tests depend upon. The test-resources for those tests now reside at src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/TypecastParenPad. Also the corresponding test-resources had to be modified accordingly to overcome having duplicate classes

rnveach commented on src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/TypecastParenPadCheckTest.java:
```diff
@@ -52,7 +54,7 @@ public void testDefault()
             "89:21: " + getCheckMessage(MSG_WS_PRECEDED, ")"),
         };
-        verify(checkConfig, getPath("InputWhitespace.java"), expected);
+        verify(checkConfig, getPath("InputWhitespace[TypecastParenPad].java"), expected);
```
Made update to original issue to explain the brackets. They were just for show and should not be in file name.

Also check name should be right after `Input` so I am expecting names like `InputTypecastParenPadWhitespace.java`, `InputTypecastParenPadAround.java`, etc.
Naming after `InputTypecastParenPad` should be similar to original name so it is easier to tell where the file came from if it was copied and not moved.

rnveach commented on src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/TypecastParenPadCheckTest.java:
```diff
@@ -40,7 +40,9 @@
         return super.getPath("checks" + File.separator
-                + "whitespace" + File.separator + filename);
+                + "whitespace" + File.separator
+                + "TypecastParenPad" + File.separator
```
I think folder names should be in lowercase like all our other package names.

romani replied:
agree. Inputs are compilable, so folders are packages for them.

rnveach commented:
@ps-sp 
Number of commits must remain as 1 for this PR to be merged. We will not merge with multiple commits.

It also looks like your commit messages are ~400 characters long and only on 1 line. Let me know if I am seeing this wrong. I'm surprised whatever git client you use didn't force a line break on you and trigger our commit validation test.
Please make commit message small and concise like PR title while keeping issue number in it.
We prefer short commit messages for easier reading. If details of change is really needed, users can open issue and read more there.

Everything else looks good.

@romani Should we enforce a maximum length on the line in `CommitValidationTest`?
A long message seems to defeat the purpose of limiting message to just 1 line.

MEZk commented:
@ps-sp 
Commit messages are too broad. Please shorten them to state the general idea, not the implementation details. Also squash commits in one as all of the changes relate to the original issue.

ps-sp commented:
@rnveach Yeah I too thought `CommitValidationTest` might fail but it didn't. Anyways, so what exactly do you suggest ? Should I create a new pull request with just one commit and a concise commit message or is there some way to edit this pull request itself ? Sorry for the inconvenience.

romani commented:
> Should we enforce a maximum length on the line in CommitValidationTest?

lets do some limit in 200 symbols (max amount of symbols that is shown on my 24" monitor in format of 'git log')
some fun statistic - `git log --pretty=format:'%s' | awk '{ print length($0); }' | sort -n | uniq`

romani commented:
>  Anyways, so what exactly do you suggest ? Should I create a new pull request with just one commit and a concise commit message or is there some way to edit this pull request itself ?

commit should have link to issue tracker and some basic information. If you need more information - put it to Issue as comment or PR as comment.

ps-sp commented:
@romani So just to be clear, I should create another pull request [y/n] ? If yes, I will be using the commit message "Issue #3900: Modified TypecastParenPadCheckTest.java and moved its test-files to a subdir" ! Would that do it ?
Thanks and apologies again for all the inconvenience.

rnveach commented:
> I should create another pull request [y/n] ? 

New PR isn't needed. Just amend/rebase current commits.
[See MEZk's comments on how to do this if your unfamiliar.](https://github.com/checkstyle/checkstyle/pull/3890#issuecomment-283462173)

> lets do some limit in 200 symbols

I will create new issue.

ps-sp commented:
@romani  @rnveach  Give it a look. Hope everything would be fine now. Thanks.
